src/ucs/memory/rcache.c

```diff
@@ -1246,10 +1246,7 @@ static UCS_CLASS_INIT_FUNC(ucs_rcache_t, const ucs_rcache_params_t *params,
         goto err_destroy_stats;
     }
 
-    status = ucs_spinlock_init(&self->lock, 0);
-    if (status != UCS_OK) {
-        goto err_destroy_rwlock;
-    }
+    ucs_spinlock_init(&self->lock);
```

Comment:
`err_destroy_rwlock` is not used anymore

Reply:
right, and there was missing spinlock cleanup on error